lib/utils/handle-type-change.js

```diff
@@ -1,9 +1,18 @@
 module.exports = handleTypeChange
 
-function handleTypeChange (emitter, type, eventName, object) {
-  if (object.type === type) {
+function handleTypeChange (state, emitter, type, eventName, object) {
+  var wasScopeType = state.objectTypeById[object.id] === type
+  var isScopeType = object.type === type
+
+  if (wasScopeType && !isScopeType) {
+    emitter.emit('remove:' + type, {id: object.id, type: type})
+  }
+  if (!wasScopeType && isScopeType) {
+    state.objectTypeById[object.id] = type
+    emitter.emit('add:' + type, object)
+  }
+  if (isScopeType || wasScopeType) {
     emitter.emit(eventName + ':' + type, object)
     emitter.emit('change:' + type, eventName, object)
   }
 }
```

Comment:
> Because what we want is that if I change a type in the global API, I want change & update events on the global API, change & remove events on scoped APIs for the old type, and change & add events on the scoped API of the new type.